Add flag to disable task rule creation

nhoughto opened the pull request.

deepy commented on src/main/kotlin/com/github/gradle/node/NodePlugin.kt:
```diff
@@ -25,8 +25,10 @@ class NodePlugin : Plugin<Project> {
         addTasks()
-        addNpmRule()
-        addYarnRule()
+        if (!nodeExtension.disableTaskRules.get()) {
```
I think we can guarantee that this works if we move this check into the rule itself, e.g. only create the task if `disableTaskRules` is `false`

nhoughto commented:
tests seem broken on `master` thus they are failing here, not much i can do about that until master is fixed?

deepy commented:
I've pushed some additional tests to master and fixed a bug in the task rules, though lately it should have been fine https://github.com/node-gradle/gradle-node-plugin/actions?query=workflow%3ABuild+branch%3Amaster

I was going to look at this and try to get it merged into master yesterday, but while out walking with my dog I fell and I've been too tired to get much done since the, hoping to be able to have a look at this tomorrow though

nhoughto commented:
Still getting failures locally
```
KotlinDsl_integTest > build project using Kotlin DSL[0] FAILED
    org.gradle.testkit.runner.UnexpectedBuildFailure: Unexpected build execution failure in /private/var/folders/w_/cwfy6zb94c94w_7hc2sw16wc0000gn/T/junit10729155429416991794 with arguments [run, --warning-mode=fail, --configuration-cache]

    Output:
    Configuration cache is an incubating feature.
    Calculating task graph as no configuration cache is available for tasks: run
    > Task :nodeSetup SKIPPED
    > Task :npmSetup SKIPPED

    > Task :npmInstall
    npm WARN deprecated debug@4.2.0: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
    npm WARN deprecated fsevents@2.1.3: Please update to v 2.2.x
    npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
    npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
    npm notice created a lockfile as package-lock.json. You should commit this file.
    npm WARN hello@ No description
    npm WARN hello@ No repository field.
    npm WARN hello@ No license field.

    added 290 packages from 167 contributors and audited 290 packages in 6.363s

    17 packages are looking for funding
      run `npm fund` for details

    found 0 vulnerabilities


    > Task :testNpm

    > hello@ test /private/var/folders/w_/cwfy6zb94c94w_7hc2sw16wc0000gn/T/junit10729155429416991794
    > mocha "test"



      hello
        #sayHello()
          ✓ should say hello


      1 passing (5ms)


    > Task :testNpx


      hello
        #sayHello()
          ✓ should say hello


      1 passing (8ms)


    > Task :yarnSetup

    > yarn@1.22.10 preinstall /private/var/folders/w_/cwfy6zb94c94w_7hc2sw16wc0000gn/T/junit10729155429416991794/.cache/yarn/yarn-latest/lib/node_modules/yarn
    > :; (node ./preinstall.js > /dev/null 2>&1 || true)

    /private/var/folders/w_/cwfy6zb94c94w_7hc2sw16wc0000gn/T/junit10729155429416991794/.cache/yarn/yarn-latest/bin/yarn -> /private/var/folders/w_/cwfy6zb94c94w_7hc2sw16wc0000gn/T/junit10729155429416991794/.cache/yarn/yarn-latest/lib/node_modules/yarn/bin/yarn.js
    /private/var/folders/w_/cwfy6zb94c94w_7hc2sw16wc0000gn/T/junit10729155429416991794/.cache/yarn/yarn-latest/bin/yarnpkg -> /private/var/folders/w_/cwfy6zb94c94w_7hc2sw16wc0000gn/T/junit10729155429416991794/.cache/yarn/yarn-latest/lib/node_modules/yarn/bin/yarn.js
    + yarn@1.22.10
    added 1 package in 0.685s

    > Task :testYarn FAILED

    FAILURE: Build failed with an exception.

    * What went wrong:
    Execution failed for task ':testYarn'.
    > A problem occurred starting process 'command 'yarn''

    * Try:
    Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

    * Get more help at https://help.gradle.org

    BUILD FAILED in 1m 36s
    5 actionable tasks: 5 executed
    Configuration cache entry stored.
        at org.gradle.testkit.runner.internal.DefaultGradleRunner$2.execute(DefaultGradleRunner.java:255)
        at org.gradle.testkit.runner.internal.DefaultGradleRunner$2.execute(DefaultGradleRunner.java:251)
        at org.gradle.testkit.runner.internal.DefaultGradleRunner.run(DefaultGradleRunner.java:324)
        at org.gradle.testkit.runner.internal.DefaultGradleRunner.build(DefaultGradleRunner.java:251)
        at com.github.gradle.AbstractIntegTest.build(AbstractIntegTest.groovy:42)
        at com.github.gradle.node.KotlinDsl_integTest.build project using Kotlin DSL(KotlinDsl_integTest.groovy:20)
```
